src/lib/parser/core/PokemonSaveParser.ts

```diff
@@ -67,19 +82,51 @@ export class PokemonSaveParser {
   private config: GameConfig | null = null
   public saveFileName: string | null = null
   public fileHandle: FileSystemFileHandle | null = null
+  private silent = false
+
+  // Memory mode properties
+  private webSocketClient: MgbaWebSocketClient | null = null
+  private isMemoryMode = false
 
-  constructor (forcedSlot?: 1 | 2, gameConfig?: GameConfig) {
+  constructor (forcedSlot?: 1 | 2, gameConfig?: GameConfig, silent?: boolean) {
     this.forcedSlot = forcedSlot
     this.config = gameConfig ?? null
+    this.silent = silent ?? false
+  }
+
+  /**
+   * Set silent mode to suppress console output
+   */
+  setSilent (silent: boolean): void {
+    this.silent = silent
   }
 
   /**
-   * Load save file data from a File or ArrayBuffer
+   * Load save file data from a File, ArrayBuffer, or WebSocket connection
+   * When WebSocket is provided, switches to memory mode
    */
-  async loadSaveFile (input: File | ArrayBuffer | FileSystemFileHandle): Promise<void> {
+  async loadSaveFile (input: File | ArrayBuffer | FileSystemFileHandle | MgbaWebSocketClient): Promise<void> {
     try {
       // Always clear sectorMap before loading new data to avoid stale state
       this.sectorMap.clear()
+
+      // Check if input is a WebSocket client for memory mode
+      // Must be an object with the specific WebSocket client properties
+      if (typeof input === 'object' &&
+          'isConnected' in input &&
+          typeof input.isConnected === 'function' &&
+          'eval' in input &&
+          typeof input.eval === 'function' &&
+          !(input instanceof ArrayBuffer) &&
+          !(input instanceof Uint8Array)) {
+        await this.initializeMemoryMode(input)
+        return
+      }
+
+      // Reset memory mode for file operations
+      this.isMemoryMode = false
+      this.webSocketClient = null
+
       let buffer: ArrayBuffer
 
       // Only check instanceof FileSystemFileHandle if it exists (browser)
```

Comment:
the name loadSaveFile is not accurate for this function anymore, lets come uö wit ha better name.
also the check for the websocket object is very ugly. we must do this better, what about instance of checks or similar